src/main/java/com/minecolonies/coremod/blocks/BlockShingleSlab.java

```diff
@@ -0,0 +1,241 @@
+        {
+            return state.withProperty(VARIANT, ShingleSlabType.TOP);
+        }
+        if(amount == THREE_CONNECTIONS)
+        {
+            if (connectors[0])
+            {
+                return state.withProperty(VARIANT, ShingleSlabType.ONE_WAY).withProperty(FACING, EnumFacing.SOUTH);
+            }
+            else if (connectors[1])
+            {
+                return state.withProperty(VARIANT, ShingleSlabType.ONE_WAY).withProperty(FACING, EnumFacing.NORTH);
+            }
+            else if (connectors[2])
+            {
+                return state.withProperty(VARIANT, ShingleSlabType.ONE_WAY).withProperty(FACING, EnumFacing.EAST);
+            }
+            return state.withProperty(VARIANT, ShingleSlabType.ONE_WAY).withProperty(FACING, EnumFacing.WEST);
+        }
+        else if(amount == TWO_CONNECTIONS)
+        {
+            if (connectors[0] && connectors[1] && !connectors[2] && !connectors[3])
+            {
+                return state.withProperty(VARIANT, ShingleSlabType.TWO_WAY).withProperty(FACING, EnumFacing.EAST);
+            }
+            else if (!connectors[0] && !connectors[1] && connectors[2] && connectors[3])
+            {
+                return state.withProperty(VARIANT, ShingleSlabType.TWO_WAY).withProperty(FACING, EnumFacing.NORTH);
+            }
+            else if(!connectors[0] && connectors[1] && connectors[2] && !connectors[3])
+            {
+                return state.withProperty(VARIANT, ShingleSlabType.CURVED).withProperty(FACING, EnumFacing.WEST);
+            }
+            else if(connectors[0] && !connectors[1] && !connectors[2] && connectors[3])
+            {
+                return state.withProperty(VARIANT, ShingleSlabType.CURVED).withProperty(FACING, EnumFacing.EAST);
+            }
+            else if(!connectors[0] && connectors[1] && !connectors[2] && connectors[3])
+            {
+                return state.withProperty(VARIANT, ShingleSlabType.CURVED).withProperty(FACING, EnumFacing.SOUTH);
+            }
+            return state.withProperty(VARIANT, ShingleSlabType.CURVED).withProperty(FACING, EnumFacing.NORTH);
+        }
+        else if(amount == ONE_CONNECTION)
+        {
+            if (!connectors[0] && !world.isAirBlock(position.west().down()))
+            {
+                return state.withProperty(VARIANT, ShingleSlabType.THREE_WAY).withProperty(FACING, EnumFacing.NORTH);
+            }
+            else if (!connectors[1] && !world.isAirBlock(position.east().down()))
+            {
+                return state.withProperty(VARIANT, ShingleSlabType.THREE_WAY).withProperty(FACING, EnumFacing.SOUTH);
+            }
+            else if (!connectors[2] && !world.isAirBlock(position.south().down()))
+            {
+                return state.withProperty(VARIANT, ShingleSlabType.THREE_WAY).withProperty(FACING, EnumFacing.WEST);
+            }
+            else if (!connectors[3] && !world.isAirBlock(position.north().down()))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 3 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)